### PR TITLE
Feature/cc 0088 display estimate price and temporary fix bluprint error.

### PR DESCRIPTION
**Pull request by Beryl-Zhou**
The estimated budget shows different total prices when color or court size changes.
The estimated Budget includes the delivery fee (based on a single price for every 1000 tiles) and installation fee (based on different court sizes). 

A Temporary Change at blueprint to fit diff env.

aa2a24ed65aa79b86a4846ffdb56564
image

f770a90ec4a02138397bf626affd295

### CODE REVIEW

File: src/utils/priceFormat.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+const priceFormat = (tilePrice: number, deliveryPrice: number, installPrice: number ) => {
+  let totalPrice = (tilePrice + deliveryPrice + installPrice).toFixed(2);
```
Comment (jennawan) [inline src/utils/priceFormat.ts:3]: if this function is named price format, then this function should only do one thing which is formatting the price, the sum of total price should not be included.

File: src/utils/priceFormat.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+const priceFormat = (tilePrice: number, deliveryPrice: number, installPrice: number ) => {
+  let totalPrice = (tilePrice + deliveryPrice + installPrice).toFixed(2);
+  let decimal = totalPrice.indexOf('.');
+  if (decimal < 0) { 
+    decimal = totalPrice.length; 
+    totalPrice += '.'; 
+  }
+  while (totalPrice.length <= decimal + 2) { 
+    totalPrice += '0'; 
+  }
```
Comment (jennawan) [inline src/utils/priceFormat.ts:11]: I'm confused about the purpose of these codes, is it for making "1234" to "1234.00"? If so, I think the .toFixed(2) already did that for you.

P.S. single-quoted, prettier did not work?

**Review by tatoMa — COMMENTED** [review]
Is there anything I do wrong?
[screen-capture (1).webm](https://user-images.githubusercontent.com/12379895/178402355-e42c8b44-0f47-4d7e-9585-8f877cc099a0.webm)

File: src/components/PriceBar/TileColorBoard.tsx [hunk 3]
```diff
@@ -1,9 +1,56 @@
...
+  const courts = useStoreSelector((state) => state.courtSize);
+  const budget = useStoreSelector((state) => state.totalPrice.budget);
+  const { data } = useGetPriceQuery(0);
+
+  useEffect(() => {
+    let tilePrice = 0;
+    let installPrice = 0;
+    let deliveryPrice = 0;
+    let totalQuantity = 0;
+    if (data) {
```
Comment (jennawan) [inline src/components/PriceBar/TileColorBoard.tsx:22]: You can remove if (data) by simply using question mark, same for the rest.   
const priceList = data?.find((item: IPriceCalculation) => !item.isDeleted);

File: src/components/PriceBar/TileColorBoard.tsx [hunk 4]
```diff
@@ -1,9 +1,56 @@
...
+          const tileList = priceList.tiles.tilePrice.find(
+            (item: ITilePrice) => item.color === tileColor
+          );
+          if (tileList) {
+            tilePrice += (tileList.price / 100) * tile.quantity;
+            totalQuantity += tile.quantity;
+          }
+        });
+        // delivery price (a fixed price per 1000 tiles)
+        deliveryPrice += (totalQuantity / 1000) * priceList.tiles.deliveryPrice;
```
Comment (jennawan) [inline src/components/PriceBar/TileColorBoard.tsx:39]: deliveryPrice += Math.ceil(totalQuantity / 1000) * priceList.tiles.deliveryPrice

File: src/utils/priceFormat.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+
+const priceFormat = (tilePrice: number, deliveryPrice: number, installPrice: number ) => {
+  let totalPrice = (tilePrice + deliveryPrice + installPrice).toFixed(2);
+  let decimal = totalPrice.indexOf('.');
+  if (decimal < 0) { 
+    decimal = totalPrice.length; 
+    totalPrice += '.'; 
+  }
```
Comment (Jzeva) [inline src/utils/priceFormat.ts:8]: When will this case happen after you have set toFixed(2)?

File: src/components/ChangeCourtSize/CourtList.tsx [hunk 6]
```diff
@@ -1,37 +1,37 @@
 const courtList = [
   {
     courtId: "62c432cfb8a9c5f61f03831f",
```
Comment (GitHubEason) [inline src/components/ChangeCourtSize/CourtList.tsx:3]: courtId should be deleted.

File: src/store/reducer/totalPriceSlice.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  reducers: {
+    changeTotalPrice: (state: TotalPriceState, action: PayloadAction<string>) => {
+      return {
+        ...state,
+        budget: action.payload,
+      };
+    },
+  },
+});
+
```
[the remark this answers is not in the document]
Comment (Beryl-Zhou) [reply src/store/reducer/totalPriceSlice.ts:23]: coz the tiles price needs to be reset when we switch the court

**Comment by Beryl-Zhou** [comment]
> Is there anything I do wrong? [screen-capture (1).webm](https://user-images.githubusercontent.com/12379895/178402355-e42c8b44-0f47-4d7e-9585-8f877cc099a0.webm)

the base API needs to be switched to local: "http://localhost:8080/v1"

**Review by Fuchih — APPROVED** [review]

**Review by jennawan — APPROVED** [review]

**Pull request merged**